### ISSUE TITLE
Attributes

### ISSUE THREAD

**Issue by thelink2012** [dcampbell24/iup-rust#15]
Tracking progress on attributes.
If you wish to implement any of those items leave a comment :)

This issue is not about method to access individual attributes on each control, instead about the general attribute system. At this moment making attributes as methods is not a priority (but will mostly like come to life later on) and thus why this is not a track of attribute as methods.

**Main:**
**Some of the items here must be skeptically taken a look to realize if they are actually needed.**
- [ ] IupSetAttribute
  - [x] IupSetAttribute
  - [x] IupSetStrAttribute
  - [ ] IupSetAttributeId
  - [ ] IupSetStrAttributeId
  - [ ] IupSetAttributeId2
  - [ ] IupSetStrAttributeId2
- [x] IupResetAttribute
- [x] IupSetAttributeHandle
- [ ] IupGetAttribute
  - [x] IupGetAttribute
  - [x] IupGetStrAttribute
  - [ ] IupGetAttributeId
  - [ ] IupGetStrAttributeId
  - [ ] IupGetAttributeId2
  - [ ] IupGetStrAttributeId2
- [x] IupGetAllAttributes
- [x] IupGetAttributeHandle
- [x] IupSetGlobal
- [x] IupSetStrGlobal
- [x] IupGetGlobal
- [x] IupGetStrGlobal

The `IupSetAttributes` `IupGetAttributes`, and `IupSetAtt` aren't going to be implemented either because they don't fit rust or to avoid over-bloat of functions that does essentially the same thing with no advantage over the simpler version.

Sub functions will not to be implemented (at least not until we start doing attribute methods), they are a major bloat that does not even provide a performance gain _(please leave a comment if you disagree with this!)_

**Comment by thelink2012** [comment]
Sub functions will not to be implemented (at least not until we start doing attribute methods), they are a major bloat that does not even provide a performance gain _(please leave a comment if you disagree with this!)_

**Comment by dcampbell24** [comment]
We don't ever need to use them. As the IUP documentation notes:

```
All the utility functions use the IupSetStrAttribute* functions internally.
```

So we can just use those and put our own safe wrappers around them when we do attribute methods.
